src/Route/RouteCollection.php

```diff
@@ -326,7 +329,17 @@ private function resolveElement(string $code): void
         $element = $this->elements[$code];
 
         if (\is_callable($element)) {
-            $this->elements[$code] = $element();
+            $resolvedElement = $element();
+            if (!$resolvedElement instanceof Route) {
+                @trigger_error(sprintf(
+                    'Element resolved by code "%s" is not instance of "%s"; This is deprecated behaviour and will be removed in 4.0.',
```

Comment:
```suggestion
                    'Element resolved by code "%s" is not instance of "%s"; This is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0.',
```

Reply:
You also need to add a Changelog for this deprecation.